server/src/main/scala/org/http4s/server/package.scala

```diff
@@ -43,4 +44,11 @@ package object server {
         .compose(AuthedRequest(authUser))
     }
   }
+
+  private[this] val messageFailureLogger = getLogger("org.http4s.server.message-failures")
+  def messageFailureHandler(req: Request): PartialFunction[Throwable, Task[Response]] = {
+    case mf: MessageFailure =>
+      messageFailureLogger.debug(mf)(s"Message failure handling request: ${req.method} ${req.pathInfo}")
```

Comment:
I wonder if you may want more info like remote IP. Or would that fit into user customization?